internal/pkg/template/templates/workloads/partials/cf/http-listener.yml

```diff
@@ -19,14 +39,23 @@ HTTPListenerRule:
       - TargetGroupArn: !Ref TargetGroup
         Type: forward
     Conditions:
-    {{- if .AllowedSourceIps}}
+      {{- if .AllowedSourceIps}}
       - Field: 'source-ip'
         SourceIpConfig:
           Values:
-          {{- range $sourceIP := .AllowedSourceIps}}
-          - {{$sourceIP}}
-          {{- end}}
-    {{- end}}
+            {{- range $sourceIP := .AllowedSourceIps}}
+            - {{$sourceIP}}
+            {{- end}}
+      {{- end}}
+      {{- if eq $.WorkloadType "Backend Service"}}
+      - Field: 'host-header'
```

Comment:
If a client today (v1.19) has a backend manifest like this:
```
http:
  path: /api
```
and uses the LB dns name to make requests `http://internal-name-123456789.region.elb.amazonaws.com/api` then due to the host-header config these requests won't match the rule. 

Looking at this fancy fancy API: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-elasticloadbalancingv2-listenerrule-hostheaderconfig.html

Should we accept the InternalALBDNS as well as an input to `Values` so that it can match both

Reply:
Great catch! I've added the internal ALB DNS to `Values` :blush:.